### ISSUE TITLE
Preprocessing

### ISSUE THREAD

**Issue by RickyyyQ** [ubicomplab/rPPG-Toolbox#390]
Thank you for analyzing the high-quality open source projects. I have a question for you. If my dataset is a dual-view video (that is, two mp4s for one physiological signal), how should I input and preprocess it? Looking forward to your reply.

**Comment by yahskapar** [comment]
I think this should be pretty straightforward if you treated this simply as a matter of modifying your custom dataloader and subsequently your custom trainer. For example:

1. Make a [custom dataloader](https://wildvisualizer.com/conversation/wildchat/1931580). In this custom dataloader, you should probably have a common ID for your videos as well as a "view" ID. For example, video 1337 with view_IDs 0, 1, and 2 for three distinct views.
2. Update your custom trainer to utilize those view IDs as expected as a part of training. 
3. For your evaluation / test set, you could work through the evaluation part of the dataset to basically go through the same procedure (e.g., treat some group of view IDs tied to a common video ID as the same test sample, effectively).

What's your goal with dual-view video? Trying to target some kind of robustness to things like lighting? Training on multiple views but at test time only evaluating on one view?

**Comment by McJackTang** [comment]
Hi @RickyyyQ , you can refer to this repo based on rPPG-Toolbox, which supports face and hand video inputs for one physiological signal.
https://github.com/thuhci/SUMS/tree/main/rppg-Toolbox_SUMS
